src/os_posix/os_fallocate.c

```diff
@@ -23,12 +23,28 @@ __posix_std_fallocate(WT_FILE_HANDLE *file_handle, WT_SESSION *wt_session, wt_of
 #if defined(HAVE_FALLOCATE)
     WT_DECL_RET;
     WT_FILE_HANDLE_POSIX *pfh;
+    WT_SESSION_IMPL *session;
 
-    WT_UNUSED(wt_session);
-
+    session = (WT_SESSION_IMPL *)wt_session;
     pfh = (WT_FILE_HANDLE_POSIX *)file_handle;
 
+    if (S2C(session)->mmap)
+        if (pfh->mmap_file_mappable && (wt_off_t)pfh->mmap_size != offset)
```

Comment:
This is done in case someone uses extend to shorten the file. Truncate does this, for example, and I assume that extend can do that too. Feel free to just use "<" if I am wrong.